app/src/main/java/com/teamwss/websoso/ui/common/customView/WebsosoSearchEditText.kt

```diff
@@ -0,0 +1,151 @@
+package com.teamwss.websoso.ui.common.customView
+
+import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.inputmethod.InputMethodManager
+import android.widget.EditText
+import android.widget.TextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.teamwss.websoso.R
+import com.teamwss.websoso.databinding.LayoutSearchBinding
+
+class WebsosoSearchEditText @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding: LayoutSearchBinding = LayoutSearchBinding.inflate(LayoutInflater.from(context), this, true)
+
+    private var externalFocusChangeListener: OnFocusChangeListener? = null
+    private var externalActionListener: TextView.OnEditorActionListener? = null
+    private var externalClearClickListener: OnClickListener? = null
+
+    init {
+        setupWebsosoSearchTextChangedListener()
+        setupWebsosoSearchClearClickListener()
+        setupWebsosoSearchFocusChangeListener()
+        setupWebsosoSearchActionListener()
+        updateSearchBackground(null, false)
+    }
+
+    /* Sets up a listener to detect changes in the text input */
+    private fun setupWebsosoSearchTextChangedListener() {
+        binding.etCommonSearch.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
```

Comment:
c: 
```suggestion
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) = Unit 
```

아래 afterTextChanged도 마찬가지!

Reply:
좋네용 깔끔해지구